inc/ThirdParty/Hostings/OneCom.php

```diff
@@ -0,0 +1,235 @@
+
+		if ( $this->is_oc_cdn_enabled() ) {
+			return $cdn;
+		}
+
+		if ( ! $cdn ) {
+			return $cdn;
+		}
+
+		// Remove CDN CNAME & CDN Zone.
+		$this->update_cdn_options( 0, [], [] );
+
+		rocket_clean_domain();
+
+		return false;
+	}
+
+	/**
+	 * Exclude files from being rewritten.
+	 *
+	 * @param array $files Array of files to be excluded.
+	 * @return array
+	 */
+	public function exclude_from_cdn( array $files ): array {
+
+		if ( ! $this->is_oc_cdn_enabled() ) {
+			return $files;
+		}
+
+		$files[] = '/wp-includes/(.*)';
+
+		return $files;
+	}
+
+	/**
+	 * Disable CDN option change.
+	 *
+	 * @param array $settings CDN field settings data.
+	 * @return array
+	 */
+	public function disable_cdn_change( array $settings ): array {
+		if ( ! $this->is_oc_cdn_enabled() ) {
+			return $settings;
+		}
+
+		$settings['cdn']['container_class'][]      = 'wpr-isDisabled';
+		$settings['cdn']['input_attr']['disabled'] = 1;
+
+		return $settings;
+	}
+
+	/**
+	 * Purge varnish cache.
+	 *
+	 * @return boolean
+	 */
+	public function should_display_varnish_auto_purge_input(): bool {
+		return ! $this->is_varnish_active();
+	}
+
+	/**
+	 * Set varnish addon title
+	 *
+	 * @param array $settings Varnish settings field data.
+	 * @return array
+	 */
+	public function maybe_set_varnish_addon_title( array $settings ): array {
+
+		// Bail out if varnish is disabled.
+		if ( ! $this->is_varnish_active() ) {
+			return $settings;
+		}
+
+		$settings['varnish_auto_purge']['title'] = sprintf(
+			// Translators: %s = Hosting name.
+				__( 'Your site is hosted on %s, we have enabled Varnish auto-purge for compatibility.', 'rocket' ),
+				'One.com'
+			);
+
+		return $settings;
+	}
+
+	/**
+	 * Check if varnish option is enabled.
+	 *
+	 * @return boolean
+	 */
+	public function is_varnish_active() {
+		return rest_sanitize_boolean( get_option( 'varnish_caching_enable' ) );
+	}
+
+	/**
+	 * Build CDN CNAME.
+	 *
+	 * @return string
+	 */
+	public function build_cname(): string {
+		if ( ! isset( $_SERVER['ONECOM_DOMAIN_NAME'] ) && ! isset( $_SERVER['HTTP_HOST'] ) ) {
+			return '';
+		}
+
+		$domain_name = sanitize_text_field( wp_unslash( $_SERVER['ONECOM_DOMAIN_NAME'] ) );
+		$http_host   = sanitize_text_field( wp_unslash( $_SERVER['HTTP_HOST'] ) );
+
+		$is_subdomain = '' === str_replace( $domain_name, '', $http_host ) ? false : true;
+
+		return $is_subdomain ? "usercontent.one/wp/$http_host" : "usercontent.one/wp/www.$http_host";
+	}
+
+	/**
+	 * Update CDN options.
+	 *
+	 * @param integer $enable_cdn Enable/Disable CDN option.
+	 * @param array   $cdn_cnames CDN CNAMES.
+	 * @param array   $cdn_zones CDN Zones.
+	 * @return void
+	 */
+	private function update_cdn_options( int $enable_cdn, array $cdn_cnames, array $cdn_zones ) {
+		$this->options->set( 'cdn', $enable_cdn );
+		$this->options->set( 'cdn_cnames', $cdn_cnames );
+		$this->options->set( 'cdn_zone', $cdn_zones );
+
+		$this->options_api->set( 'settings', $this->options->get_options() );
```

Comment:
Every time any request happen, we will update those settings and we don't really need that.

You can easily use the filter `pre_get_rocket_option_OPTIONKEY` that is defined here: 
https://github.com/wp-media/wp-rocket/blob/5434080ccd89b36f3312846e4288a50c9d6b7c3e/inc/classes/admin/class-options-data.php#L58

to return the values u need without changing the database.
or if u continued in this update approach u should find a way to make this update only when needed, not every time.

what do u think?